generate-errors/src/lib.rs

```diff
@@ -1,114 +1,213 @@
+                ))?
+                .split(',');
+
+            let mut line: String = String::from("```");
+
+            // Add all annotations other than rust
+            // to the content first to avoid the issue.
+            for annotation in annotations {
+                if annotation == "rust" {
+                    continue;
+                }
+
+                line.push_str(annotation);
+                line.push(',');
+            }
+            line.push_str("rust");
+
+            content.push_str(&line);
+            content.push('\n');
         }
+
+        results_map.insert(file_name, content.to_string());
     }
 
-    section.content.sort_by(|a, b| a.code.cmp(&b.code));
+    Ok(results_map)
+}
+
+/// Writes a valid docs section to contain
+/// the error pages in.
+///
+/// The output path passed should be the folder
+/// you want the Zola pages to be written / output.
+pub fn write_section(output_path: &Path) -> anyhow::Result<()> {
+    let errors_folder_path = output_path.join("errors");
+    // Make sure the output folder exists
+    fs::create_dir_all(&errors_folder_path)?;
+
+    let section_content = r#"+++
+title = "Errors"
+template = "docs-section.html"
+page_template = "docs-page.html"
+redirect_to = "/learn/errors/introduction"
++++
+"#;
+
+    let introduction_content = r#"+++
+title = "Introduction"
+[extra]
+weight = 0
++++
+
+These pages document Bevy's error codes for the _current release_.
+
+In case you are looking for the latest error codes from Bevy's main branch, you can find them in the [Bevy engine repository](<https://github.com/bevyengine/bevy/tree/main/errors>). 
+"#;
+
+    fs::write(errors_folder_path.join("_index.md"), section_content)?;
+    fs::write(
+        errors_folder_path.join("introduction.md"),
+        introduction_content,
+    )?;
 
     Ok(())
 }
 
-pub fn parse_errors(errors_dir: &str) -> io::Result<Section> {
-    let mut errors_root_section = Section {
-        name: "Errors".to_string(),
-        content: vec![],
-        template: Some("errors.html".to_string()),
-        header: Some("Errors".to_string()),
-        order: None,
-        sort_order_reversed: false,
-    };
-    visit_dirs(
-        PathBuf::from_str(errors_dir).unwrap(),
-        &mut errors_root_section,
-    )?;
-    Ok(errors_root_section)
+pub fn write_pages(output_path: &Path, pages: HashMap<String, String>) -> anyhow::Result<()> {
+    let errors_folder_path = output_path.join("errors");
+    // Make sure the output folder exists
+    fs::create_dir_all(&errors_folder_path)?;
+
+    // Make the keys ordered so that
+    // we know the weights for the pages
+    let mut keys: Vec<&String> = pages.keys().collect();
+    keys.sort_unstable();
+
+    for (index, key) in keys.iter().enumerate() {
+        let page_content = format!(
+            r#"+++
+title = "{}"
+[extra]
+weight = {}
++++
+
+{}"#,
+            key.strip_suffix(".md").unwrap_or(key),
```

Comment:
This could use a comment explaining how it is extracting the error code from the filename, and just using the filename for the title if stripping fails.